homeassistant/components/swidget/entity.py

```diff
@@ -0,0 +1,39 @@
+"""Common code for Swidget."""
+
+from __future__ import annotations
+
+from swidget.swidgetdevice import SwidgetDevice
+
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import SwidgetDataUpdateCoordinator
+
+
+class CoordinatedSwidgetEntity(CoordinatorEntity[SwidgetDataUpdateCoordinator]):
+    """Common base class for all coordinated entities."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, device: SwidgetDevice, coordinator: SwidgetDataUpdateCoordinator
+    ) -> None:
+        """Initialize the Swidget device."""
+        super().__init__(coordinator)
+        self.device: SwidgetDevice = device
+        self._attr_name = self.device.friendly_name
```

Comment:
Device name is already this, so this should be `_attr_name = None`

Reply:
Fixed.